docs/adr/0000-use-es-compatible-unit-test-frameworks.md

```diff
@@ -0,0 +1,174 @@
+# Use ES-compatible unit test frameworks
+
+* Status: proposed
+* Deciders:
+* Date: 2023-01-17
+
+Technical Story: [description | ticket/issue URL] <!-- optional -->
+
+## Context and Problem Statement
+
+The Firefox Monitor "blurts" server is catching up on technical debt as part of
+a move from V1 to V2:
+
+- reducing dependencies where feasible/adivsable to reduce the risk of [supply-chain attacks](https://snyk.io/blog/npm-security-preventing-supply-chain-attacks/)
+- move from the older [CommonJS module system](https://nodejs.org/api/modules.html) to the more standardized [ECMAScript Modules (ES aka ESM)](https://nodejs.org/api/esm.html)
+
+Monitor V1 uses the [Jest](https://jestjs.io/) for unit tests. Jest is both a test runner and also a [mocking](https://jestjs.io/docs/mock-functions) library.
+V1 currently has [~30% code coverage](https://coveralls.io/builds/55941931) over 24 tests, and uses Jest mocks in 11 of
+these to mock 8 modules.
+
+Jest does not fully support ES6+ modules for several reasons:
+
+* Per the spec, ES6+ does not allow so-called "monkey-patching" already-loaded modules
+* Jest expects to be able to "hoist" it's `jest.mock` commands before modules are loaded, which also isn't allowed in ES6+
+* Node still requires an experimental flag to load ES modules as of this writing
```

Comment:
I'm not sure this line about ESM experimental flag is true?  From this page: https://nodejs.org/docs/latest-v18.x/api/esm.html#modules-ecmascript-modules
>Node.js fully supports ECMAScript modules as they are currently specified and provides interoperability between them and its original module format, [CommonJS](https://nodejs.org/docs/latest-v18.x/api/modules.html).

Reply:
Jest needs this specifically: `--experimental-vm-modules` https://nodejs.org/api/vm.html#vm_class_vm_module

Reply:
OK I clarified which feature this is referring to, it's really that Jest (when running in pure ES mode) requires an experimenta